src/execution_plan/ops/op_delete.c

```diff
@@ -1,253 +1,146 @@
-// #include "./op_delete.h"
-
-// /* Forward declarations. */
-// void _LocateEntities(OpDelete *op_delete, QueryGraph *graph, AST_DeleteNode *ast_delete_node);
-// void _DeleteEdge(OpDelete *op, Edge *e);
-
-// OpBase* NewDeleteOp(RedisModuleCtx *ctx, AST_DeleteNode *ast_delete_node, QueryGraph *graph,
-//                     const char *graph_name, ResultSet *result_set) {
-//     return (OpBase*)_NewDeleteOp(ctx, ast_delete_node, graph, graph_name, result_set);
-// }
-
-// OpDelete* _NewDeleteOp(RedisModuleCtx *ctx, AST_DeleteNode *ast_delete_node, QueryGraph *graph,
-//                        const char *graph_name, ResultSet *result_set) {
-//     OpDelete *op_delete = (OpDelete*) calloc(1, sizeof(OpDelete));
-
-//     op_delete->ctx = ctx;
-//     op_delete->request_refresh = 1;
-//     op_delete->graph_name = graph_name;    
-//     op_delete->nodes_to_delete = (Node***)malloc(sizeof(Node**) * Vector_Size(ast_delete_node->graphEntities));
-//     op_delete->node_count = 0;
-//     op_delete->edges_to_delete = (Edge***)malloc(sizeof(Edge**) * Vector_Size(ast_delete_node->graphEntities));
-//     op_delete->edge_count = 0;
-//     op_delete->deleted_nodes = NewTrieMap();
-//     op_delete->deleted_edges = NewTrieMap();
-//     op_delete->result_set = result_set;
+#include "./op_delete.h"
+
+/* Forward declarations. */
+void _LocateEntities(OpDelete *op_delete, QueryGraph *graph, AST_DeleteNode *ast_delete_node);
+
+OpBase* NewDeleteOp(AST_DeleteNode *ast_delete_node, QueryGraph *qg, Graph *g, ResultSet *result_set) {
+    return (OpBase*)_NewDeleteOp(ast_delete_node, qg, g, result_set);
+}
+
+OpDelete* _NewDeleteOp(AST_DeleteNode *ast_delete_node, QueryGraph *qg, Graph *g, ResultSet *result_set) {
+    OpDelete *op_delete = (OpDelete*) calloc(1, sizeof(OpDelete));
```

Comment:
Every member of the struct is explicitly set, so malloc should be fine.

Reply:
👍